bouncerscript/test/test_script.py

```diff
@@ -40,28 +42,68 @@ def test_main(submission_context, fake_session):
 # bouncer_submission {{{1
 @pytest.mark.asyncio
 async def test_bouncer_submission(submission_context, mocker):
-    mocker.patch.object(bscript, 'does_product_exist', new=noop_async)
+    mocker.patch.object(bscript, 'does_product_exist', new=toggled_boolean_async)
     mocker.patch.object(bscript, 'api_add_product', new=noop_async)
     mocker.patch.object(bscript, 'api_add_location', new=noop_async)
     mocker.patch.object(bscript, 'get_locations_paths', new=noop_async)
+    mocker.patch.object(bscript, 'does_location_path_exist', new=toggled_boolean_async)
+    mocker.patch.object(bscript, 'check_locations_match', new=raise_sync)
+    mocker.patch.object(bscript, 'check_path_matches_destination', new=noop_sync)
     with pytest.raises(ScriptWorkerTaskException):
         await bouncer_submission(submission_context)
 
-    mocker.patch.object(bscript, 'does_product_exist', new=toggled_boolean_async)
-    mocker.patch.object(bscript, 'check_locations_match', new=return_false_sync)
-    mocker.patch.object(bscript, 'check_path_matches_destination', new=return_true_sync)
+    mocker.patch.object(bscript, 'check_locations_match', new=noop_sync)
+    await bouncer_submission(submission_context)
+
+    mocker.patch.object(bscript, 'check_path_matches_destination', new=raise_sync)
     with pytest.raises(ScriptWorkerTaskException):
         await bouncer_submission(submission_context)
 
-    mocker.patch.object(bscript, 'check_locations_match', new=return_true_sync)
-    await bouncer_submission(submission_context)
-
-    mocker.patch.object(bscript, 'check_locations_match', new=return_false_sync)
-    mocker.patch.object(bscript, 'check_path_matches_destination', new=return_false_sync)
+    mocker.patch.object(bscript, 'does_product_exist', new=return_false_async)
     with pytest.raises(ScriptWorkerTaskException):
         await bouncer_submission(submission_context)
 
 
+@pytest.mark.asyncio
+async def test_bouncer_submission_creates_locations_even_when_product_already_exists(
+    submission_context, mocker
+):
+    api_add_location_call_counter = (i for i in range(11))
```

Comment:
You can use `itertools.count()`  instead, which doesn't require specifying an upper limit.

Reply:
TIL! Done